jaxrs/pom.xml

```diff
@@ -62,6 +62,10 @@
             <groupId>io.swagger</groupId>
             <artifactId>swagger-annotations</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.swagger</groupId>
+            <artifactId>swagger-jersey-jaxrs</artifactId>
```

Comment:
FYI, it looks like this pulls a few `com.sun.jersey` dependencies unfortunately, but it doesn't look like there's a version conflict.

What is it for?

Reply:
Good question: I think what i really need is the following:

```
        <dependency>
            <groupId>io.swagger</groupId>
            <artifactId>swagger-jaxrs</artifactId>
            <version>1.5.18</version>
        </dependency>
        <dependency>
            <groupId>io.swagger</groupId>
            <artifactId>swagger-models</artifactId>
            <version>1.5.18</version>
        </dependency>

```

Which pulls the following:

```
[INFO] +- io.swagger:swagger-jaxrs:jar:1.5.18:compile
[INFO] |  +- io.swagger:swagger-core:jar:1.5.18:compile
[INFO] |  |  +- org.apache.commons:commons-lang3:jar:3.2.1:compile
[INFO] |  |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.9.3:compile
[INFO] |  |  |  \- org.yaml:snakeyaml:jar:1.18:compile
[INFO] |  |  \- javax.validation:validation-api:jar:1.1.0.Final:compile
[INFO] |  \- org.reflections:reflections:jar:0.9.11:compile
[INFO] |     \- org.javassist:javassist:jar:3.19.0-GA:compile (version managed from 3.21.0-GA)
[INFO] +- io.swagger:swagger-models:jar:1.5.18:compile
```

This is needed for all the pre/post processing [swagger hooks](https://github.com/killbill/killbill/blob/work-for-release-0.19.x-doc/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/KillBillApiDefinition.java#L21)

Reply:
⚠️ This won't work unfortunately as it pulls Jackson 2.9 (See also https://github.com/killbill/killbill-oss-parent/blob/master/pom.xml#L114-L115.). You can see the conflicts in the CircleCI build:

```
[ERROR] Found a problem with the direct dependency com.fasterxml.jackson.core:jackson-databind of the current project
  Expected version is 2.6.7
  Resolved version is 2.6.7
  Version 2.6.7 was expected by artifacts: org.kill-bill.billing:killbill-catalog, org.kill-bill.billing:killbill-internal-api, org.kill-bill.billing:killbill-overdue, org.kill-bill.billing:killbill-platform-server:classes, org.kill-bill.billing:killbill-platform-test, org.kill-bill.billing:killbill-util, org.kill-bill.billing:killbill-util:test-jar, org.kill-bill.commons:killbill-queue, org.kill-bill.commons:killbill-queue:test-jar
  Version 2.9.3 was expected by artifact: io.swagger:swagger-jaxrs
[ERROR] Found a problem with the direct dependency com.fasterxml.jackson.core:jackson-annotations of the current project
  Expected version is 2.6.7
  Resolved version is 2.6.7
  Version 2.6.0 was expected by artifacts: com.fasterxml.jackson.core:jackson-databind, org.kill-bill.billing:killbill-platform-server:classes
  Version 2.6.7 was expected by artifacts: org.kill-bill.billing.plugin:killbill-plugin-api-notification, org.kill-bill.billing:killbill-api, org.kill-bill.billing:killbill-catalog, org.kill-bill.billing:killbill-internal-api, org.kill-bill.billing:killbill-overdue, org.kill-bill.billing:killbill-platform-test, org.kill-bill.billing:killbill-util, org.kill-bill.billing:killbill-util:test-jar, org.kill-bill.commons:killbill-queue, org.kill-bill.commons:killbill-queue:test-jar, org.kill-bill.commons:killbill-xmlloader
  Version 2.9.3 was expected by artifacts: io.swagger:swagger-jaxrs, io.swagger:swagger-models
[ERROR] Found a problem with the direct dependency com.fasterxml.jackson.core:jackson-core of the current project
  Expected version is 2.6.7
  Resolved version is 2.6.7
  Version 2.6.7 was expected by artifacts: com.fasterxml.jackson.core:jackson-databind, org.kill-bill.billing:killbill-catalog, org.kill-bill.billing:killbill-internal-api, org.kill-bill.billing:killbill-overdue, org.kill-bill.billing:killbill-platform-server:classes, org.kill-bill.billing:killbill-platform-test, org.kill-bill.billing:killbill-util, org.kill-bill.billing:killbill-util:test-jar, org.kill-bill.commons:killbill-queue, org.kill-bill.commons:killbill-queue:test-jar
  Version 2.9.3 was expected by artifact: io.swagger:swagger-jaxrs
[ERROR] Found a problem with the direct dependency io.swagger:swagger-annotations of the current project
  Expected version is 1.5.10
  Resolved version is 1.5.10
  Version 1.5.18 was expected by artifacts: io.swagger:swagger-jaxrs, io.swagger:swagger-models
[ERROR] Found a problem with the dependency org.javassist:javassist
  Resolved version is 3.19.0-GA
  Version 3.19.0-GA was expected by artifact: org.kill-bill.billing:killbill-platform-server:classes
  Version 3.21.0-GA was expected by artifact: io.swagger:swagger-jaxrs
```

Can you downgrade to `1.5.10`? The alternative is to tackle the Jackson upgrade but it sounds like a distraction at this point.

From quickly glancing at the release notes, things to watch out for:

```
ObjectMapper default timezone now UTC, not GMT (as with 2.6 and prior): usually the two behave the same, but may cause minor variation in serialized form

SnakeYAML dependency not shaded any more (see #31); also now wraps all SnakeYAML exceptions as regular Jackson exceptions

As per databind#219 java.sql.Date will finally use same "timestamp-or-String" determination as java.util.Date and java.util.Calendar. This means that with vanilla, unchanged settings, values will be serialized as numeric timestamps. Note that the default String serialization will still default to java.sql.Date.String(), and not to default formatting java.util.Date uses.
```